src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```diff
@@ -70,6 +71,7 @@ public SimpleElasticsearchPersistentEntity(TypeInformation<T> typeInformation) {
 					" Unknown indexName. Make sure the indexName is defined. e.g @Document(indexName=\"foo\")");
 			this.indexName = typeInformation.getType().getAnnotation(Document.class).indexName();
 			this.indexType = hasText(document.type()) ? document.type() : clazz.getSimpleName().toLowerCase(Locale.ENGLISH);
+			this.serverConfig = typeInformation.getType().getAnnotation(Document.class).serverConfig();
 			this.shards = typeInformation.getType().getAnnotation(Document.class).shards();
 			this.replicas = typeInformation.getType().getAnnotation(Document.class).replicas();
 			this.refreshInterval = typeInformation.getType().getAnnotation(Document.class).refreshInterval();

```

Comment:
These `typeInformation.getType().getAnnotation(Document.class)` could be cached in a local variable. No need to re-read several times.